src/main/java/com/github/princesslana/slothbot/Discord.java

```diff
@@ -15,11 +21,33 @@ public static String getGuildId(DiscordRequest req) {
     return req.getDispatcher().guildFromEvent(req.getEvent());
   }
 
+  public static Optional<String> getGuild(com.google.gson.JsonObject payload) {
+    return Optional.ofNullable(payload.get("guild_id")).map(JsonElement::getAsString);
+  }
+
+  public static Set<String> getRoles(com.google.gson.JsonObject payload) {
+    return Optional.ofNullable(payload.get("member"))
+        .map(JsonElement::getAsJsonObject)
+        .map(inner -> inner.get("roles"))
+        .map(JsonElement::getAsJsonArray)
+        .map(JsonArray::iterator)
+        .map(Discord::iteratorToSet)
```

Comment:
Not being able to find this method in the JDK bugs me.
It does exist in guava tho with ImmutableSet.copyOf, or Sets.newHashSet